### PR TITLE
Dashboard layout

### PR DESCRIPTION
**Pull request by justin-cook-developer**
Refactored the CardGroup to be a general component for making card groups. Created an initial layout for the Dashboard.

### CODE REVIEW

File: client/components/Dashboard/DashboardDumb.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    name: 'Algorithms',
+    description: 'All things Algorithms!',
+  },
+  {
+    id: 6,
+    name: 'Data Structures',
+    description: 'All things Data Structures!',
+  },
+];
+
```
Comment (nicknapoli82) [inline client/components/Dashboard/DashboardDumb.js:38]: Now I know where the things came from in the snapshot you showed us in our slack channel.

Comment (justin-cook-developer) [reply client/components/Dashboard/DashboardDumb.js:38]: Yep! Just made some dummy data.

File: utils/index.js [hunk 2]
```diff
@@ -21,8 +21,14 @@ const titleCase = str => {
   }
 };
 
+const firstSentenceOnly = p => {
+  const idx = p.search(/(\?|\.|!)/);
+  return p.slice(0, idx + 1);
+};
+
```
Comment (nicknapoli82) [inline utils/index.js:33]: I like your line of thinking here, but I'm not sure if you can rely on the fact that a sentence will exist proper in whatever a user chooses to put in a particular field. @CB6670 mentioned the idea of tourniqueting the length of undetermined user input is just to take a slice of the string up to a defined point and append '...' to the end. I kind of agree with him in this case.
return str.slice(0,25) + '...';

Comment (justin-cook-developer) [reply utils/index.js:33]: What about attempting to find the index of a sentence ending punctuation mark, then if it does not exist, using a predetermined length?

**Review by nicknapoli82 — DISMISSED** [review]
I'm approving here. Everything looks good here @justin-cook-developer with exception of the one comment I made in the client/utils file

File: utils/index.js (hunk 2 above)
Comment (nicknapoli82) [reply utils/index.js:33]: I think that would totally work also man.

Comment (justin-cook-developer) [reply utils/index.js:33]: Or is the worry that a punctuation mark might occur too far into the text?

Comment (nicknapoli82) [reply utils/index.js:33]: Maybe to far in the text, maybe there is no punctuation at all but a really long body of text.

**Review by nicknapoli82 — APPROVED** [review]
Awesome @justin-cook-developer

**Pull request merged**